### ISSUE TITLE
MA0192 is not reported when "comparand" is not a constant

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#1313]
### Version of the Meziantou.Analyzer NuGet package

3.0.173

### Rule Identifier

MA0192 

### Target Framework

.NET 10.0

### C# Language version

C# 14

### .NET SDK version

10.0.400

### Description

**MA0192** is not reported when the "comparand" of the bitwise comparison is not a constant.

### Reproduction Steps

Minimal code:

```c#
using System.Reflection.Emit;

private static bool Param(AssemblyBuilderAccess comparand)
{
    var value = AssemblyBuilderAccess.Run | AssemblyBuilderAccess.RunAndCollect;
    return (value & comparand) == comparand;
}
```

**Comment by drieseng** [comment]
**Note:**
Looking at the [implementation](https://github.com/dotnet/runtime/blob/1967cbeed831dcff79d99b59e20c55ab8b5c9fc0/src/libraries/System.Private.CoreLib/src/System/Enum.cs#L399), I was hesitant to accept your "advice".
Benchmarks did show performance is on par with bitwise comparison though.
